eng/scripts/InstallVisualStudio.ps1

```diff
@@ -59,29 +65,36 @@ mkdir $intermedateDir -ErrorAction Ignore | Out-Null
 $bootstrapper = "$intermedateDir\vsinstaller.exe"
 $ProgressPreference = 'SilentlyContinue' # Workaround PowerShell/PowerShell#2138
 
-$channelUri = "https://aka.ms/vs/16/release"
-$responseFileName = "vs"
+if ("$Version" -eq "2019") {
+    $vsversion = "16";
+}
+if ("$Version" -eq "2022") {
+    $vsversion = "17";
+    $Channel = "Preview";
+}
+$channelUri = "https://aka.ms/vs/$vsversion/release"
+$responseFileName = "vs.$vsversion"
 if ("$Edition" -eq "BuildTools") {
     $responseFileName += ".buildtools"
 }
 if ("$Channel" -eq "Preview") {
     $responseFileName += ".preview"
-    $channelUri = "https://aka.ms/vs/16/pre"
+    $channelUri = "https://aka.ms/vs/$vsversion/pre"
 }
 
 $responseFile = "$PSScriptRoot\$responseFileName.json"
 $channelId = (Get-Content $responseFile | ConvertFrom-Json).channelId
 
 $bootstrapperUri = "$channelUri/vs_$($Edition.ToLowerInvariant()).exe"
-Write-Host "Downloading Visual Studio 2019 $Edition ($Channel) bootstrapper from $bootstrapperUri"
+Write-Host "Downloading Visual Studio $Version $Edition ($Channel) bootstrapper from $bootstrapperUri"
 Invoke-WebRequest -Uri $bootstrapperUri -OutFile $bootstrapper
 
 $productId = "Microsoft.VisualStudio.Product.$Edition"
 if (-not $InstallPath) {
     $vsWhere = "${env:ProgramFiles(x86)}\Microsoft Visual Studio\Installer\vswhere.exe"
     if (Test-Path $vsWhere)
     {
-        $installations = & $vsWhere -version '[16,17)' -format json -prerelease -products $productId | ConvertFrom-Json |Sort-Object -Descending -Property installationVersion, installDate
+        $installations = & $vsWhere -version '[16,18)' -format json -prerelease -products $productId | ConvertFrom-Json |Sort-Object -Descending -Property installationVersion, installDate
```

Comment:
This will mean 2022 will overwrite 2019. Suggest
```suggestion
        $installations = & $vsWhere -version "[$vsversion,$($vsversion+1))" -format json -prerelease -products $productId | ConvertFrom-Json |Sort-Object -Descending -Property installationVersion, installDate
```

Reply:
would need to be plus 2 yes, due to the `[ )` range limiter?

Reply:
nevermind, brain slow

Reply:
No, current `[16,17)` works fine for 2019 and `[17,18}` will be fine for 2022

Reply:
IOW I did the `pwsh` right 😀